middlewares/auth_required.js

```diff
@@ -1,3 +1,13 @@
-/***
- * @todo Redirect the user to login page if token is not present.
- */
+import { useAuth } from "../context/auth";
+import { useRouter } from "next/router";
+
+const { token } = useAuth();
+const router = useRouter();
+
+console.log("token logged in auth_req");
+console.log(token);
+
+if(token === undefined || token === null)
+{
+    router.push("../pages/login.js")
```

Comment:
Incorrect way to do routing

Reply:
I am not sure what you meant. Is there any problem with router.push?